File: ios/Sources/EmarsysPlugin/EmarsysConfig.swift
import Foundation
import EmarsysSDK

@objc public class EmarsysConfig: NSObject {
    @objc public func changeApplicationCode(_ applicationCode: String, completion: @escaping (Error?) -> Void) {
        Emarsys.config.changeApplicationCode(applicationCode: applicationCode) { error in
            completion(error)
        }
    }

    @objc public func changeMerchantId(_ merchantId: String) {
        Emarsys.config.changeMerchantId(merchantId: merchantId)
    }

    @objc public func getApplicationCode() -> String? {
        return Emarsys.config.applicationCode()
    }

    @objc public func getMerchantId() -> String? {
        return Emarsys.config.merchantId()
    }

    @objc public func getClientId() -> String? {
        return Emarsys.config.clientId()
    }

    @objc public func getLanguageCode() -> String? {
        return Emarsys.config.languageCode()
    }

    @objc public func getSdkVersion() -> String? {
        return Emarsys.config.sdkVersion()
    }
}

File: android/src/main/java/com/emarsys/plugin/emarsys/EmarsysConfig.java
package com.emarsys.plugin.emarsys;

import com.emarsys.Emarsys;

public class EmarsysConfig {

    public void changeApplicationCode(String applicationCode, EmarsysCompletionListener listener) {
        Emarsys.getConfig().changeApplicationCode(applicationCode, error -> listener.onComplete(error));
    }

    public void changeMerchantId(String merchantId) {
        Emarsys.getConfig().changeMerchantId(merchantId);
    }

    public String getApplicationCode() {
        return Emarsys.getConfig().getApplicationCode();
    }

    public String getMerchantId() {
        return Emarsys.getConfig().getMerchantId();
    }

    public String getClientId() {
        return Emarsys.getConfig().getClientId();
    }

    public String getLanguageCode() {
        return Emarsys.getConfig().getLanguageCode();
    }

    public String getSdkVersion() {
        return Emarsys.getConfig().getSdkVersion();
    }
}
